### PR TITLE
Add signal which send data with old and new value of address and host…

### PR DESCRIPTION
**Pull request by ar4s**
Related to https://github.com/allegro/django-powerdns-dnssec/pull/146

### CODE REVIEW

File: src/ralph/lib/mixins/models.py [hunk 1]
```diff
@@ -102,3 +102,15 @@ class TaggableMixin(models.Model):
 
     class Meta:
         abstract = True
+
+
+class PreviousStateMixin(models.Model):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        fields = self._meta.get_all_field_names()
```
Comment (ghost) [inline src/ralph/lib/mixins/models.py]: https://docs.djangoproject.com/en/1.9/ref/models/meta/#migrating-from-the-old-api

Comment (ghost) [reply src/ralph/lib/mixins/models.py]: [f.name for f in MyModel._meta.get_fields()]

File: src/ralph/networks/receivers.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    data_to_send = {
+        'new': {
+            key: instance.__dict__[key] for key in keys
+        },
+        'old': {
+            key: instance._previous_state[key] for key in keys
+        }
+    }
+    data_to_send['action'] = 'add' if created else 'update'
+    DNSaaS().send_ipaddress_data(data_to_send)
```
Comment (mkurek) [inline src/ralph/networks/receivers.py]: do we want to send "event" even when hostname is empty?

File: src/ralph/dns/dnsaas.py [hunk 3]
```diff
@@ -216,3 +216,8 @@ def delete_dns_record(self, record_id):
             }
         elif response.status_code != 204:
             return response.json()
+
+    def send_ipaddress_data(self, update_data):
```
Comment (ghost) [inline src/ralph/dns/dnsaas.py]: Perhaps some info about `update_data` could be added.

**Comment by ghost** [comment]
:ok:

**Comment by mkurek** [comment]
could you write some tests (mocking sending data to dnsaas (but it'd perfect to have integration tests for that in the future))? ex. for adding new ip, updating ip, deleting ip, deleting server (ex. DataCenterAsset or VirtualServer)

File: src/ralph/dns/dnsaas.py [hunk 4]
```diff
@@ -216,3 +216,35 @@ def delete_dns_record(self, record_id):
...
+                ``old`` and ``new`` contains previous and next state of
+                IP address record (in Ralph).
+        Returns:
+            JSON response from API
+        """
+        logger.info('Send update data: {}'.format(ip_record_data))
+        url = self.build_url('ip_record')
+        response = self._post(url, ip_record_data)
+        if response.status_code >= 400:
+            logger.info('DNSaaS returned {}'.format(response.status_code))
```
Comment (szok) [inline src/ralph/dns/dnsaas.py]: maybe 'warning' or 'error' should be better?

**Review by szok — APPROVED** [review]

**Pull request merged**